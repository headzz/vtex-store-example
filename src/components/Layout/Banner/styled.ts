import { colors } from 'configuration'
import styled from 'styled-components'

export const BannerStyled = styled.section`
  background-color: ${colors.GREEN_JUSSI};
  padding: 4rem 0;
  width: 100%;

  #banner-image {
    display: none;
  }

  @media (min-width: 768px) {
    #banner-image {
      display: block;
    }
  }
`
