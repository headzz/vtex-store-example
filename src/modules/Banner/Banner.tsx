import Flexbox from 'components/Flexbox/Flexbox'
import Layout from 'components/Layout'

import LeftArea from './components/LeftArea'
import RightArea from './components/RightArea'

/**
 * @component
 * @name Banner
 * @description creates Banner component section to use on screen.
 * @namespace components modules/Banner
 *
 */

const Banner: React.FC = () => (
  <Layout.Banner>
    <Flexbox alignContent="center">
      <LeftArea />
      <RightArea />
    </Flexbox>
  </Layout.Banner>
)

export default Banner
