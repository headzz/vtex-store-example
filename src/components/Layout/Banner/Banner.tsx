import { BannerStyled } from './styled'

/**
 * @component
 * @name Banner
 * @description Defines the container for the layout of banner.
 * @namespace components components/Layout/Banner
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */

const Banner: React.FC = ({ children }) => {
  return <BannerStyled>{children}</BannerStyled>
}

export default Banner
