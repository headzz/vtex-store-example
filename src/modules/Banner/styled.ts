import styled from 'styled-components'

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 4rem;
  position: relative;
  width: 90%;

  p:first-child {
    left: 0.8rem;
    position: absolute;
    top: 0;
  }
  p:nth-child(-n + 2) {
    font-size: 3.2rem;
  }

  p:nth-child(3) {
    margin: 1.6rem 0 2.4rem 0;
    width: 100%;
  }

  @media (min-width: 768px) {
    padding: 0;
    width: 44.8rem;

    p:first-child {
      left: -8rem;
      top: 0;
    }

    p:nth-child(3) {
      margin: 1.6rem 0 4.4rem 0;
      line-height: 2.4rem;
      width: 32.8rem;
    }

    p:nth-child(-n + 2) {
      font-size: 6.4rem;
    }

    a {
      width: fit-content;
    }
  }
`
