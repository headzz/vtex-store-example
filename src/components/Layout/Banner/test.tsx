import { render } from '@testing-library/react'

import Banner from './Banner'

describe('<Banner />', () => {
  it('should render component that wraps banner on page', () => {
    const { container } = render(<Banner />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
