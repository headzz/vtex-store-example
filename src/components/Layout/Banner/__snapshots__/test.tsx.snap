// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render component that wraps banner on page 1`] = `
.c0 {
  background-color: #03FFA5;
  padding: 4rem 0;
  width: 100%;
}

.c0 #banner-image {
  display: none;
}

@media (min-width:768px) {
  .c0 #banner-image {
    display: block;
  }
}

<section
  class="c0"
/>
`
